src/shadow-embed.js

```diff
@@ -548,6 +548,31 @@ export class ShadowDomWriterStreamer {
     return this.success_;
   }
 
+  /** @override */
+  get closed() {
+    return this.eof_;
+  }
+
+  /** @override */
+  get desiredSize() {
+    return 1; // noop
```

Comment:
Ok. As long as it actually doesn't start writing one character at a time :) Just for kicks, can you test what it actually does? Otherwise, LGTM